Add TLSRoute type listener (Gateway API support) (#12747)
strimzi/strimzi-kafka-operator: documentation/modules/overview/con-configuration-points-listeners.adoc

+
* `nodeport` to use ports on Kubernetes nodes
* `loadbalancer` to use loadbalancer services

> * `tlsroute` to use Gateway API `v1` `TLSRoute` resources

Reviewer: ```suggestion
* `tlsroute` to use Kubernetes Gateway API `v1` `TLSRoute` resources
```

* `ingress` to use Kubernetes `Ingress` and the {NginxIngressController} (Kubernetes only)
* `route` to use OpenShift `Route` and the default HAProxy router (OpenShift only)
+
External listeners handle access to a Kafka cluster from networks that require different authentication mechanisms.
For example, loadbalancers might not be suitable for certain infrastructure, such as bare metal, where node ports provide a better option.

> NOTE: The `ingress` listener type is deprecated. Consider using the `tlsroute` listener type when you have a Gateway API implementation that supports `TLSRoute` resources.

Reviewer: ```suggestion
NOTE: The `ingress` listener type is deprecated. Consider using the `tlsroute` listener type if your Gateway API implementation supports `TLSRoute` resources.
```
